### PR TITLE
Use csr-lite to resolve OpenMP race condition

### PR DESCRIPTION
**Pull request by justsz**

### CODE REVIEW

File: src/linear_solvers/custom_matrix.f90 [hunk 1]
```diff
@@ -10,39 +12,100 @@ module custom_matrix
...
+
+  subroutine zeros(matrix)
+
+    type(csr_matrix), intent(inout) :: matrix
+
+    matrix%values(:) = 0.0_ccs_real
+
+  end subroutine
+
+  subroutine get_rows_array(rows, matrix)
```
Comment (pbartholomew08) [inline src/linear_solvers/custom_matrix.f90]: Argument order is reversed

File: src/linear_solvers/custom_matrix.f90 [hunk 2]
```diff
@@ -10,39 +12,100 @@ module custom_matrix
...
+    matrix%values(start : end) = vals(:)
+    matrix%columns(start : end) = cols(:)
+
+  end subroutine
+
+  subroutine add_values(row, vals, cols, matrix)
+
+    integer(ccs_int), intent(in) :: row
+    real(ccs_real), dimension(:) :: vals
+    integer(ccs_int), dimension(:) :: cols
```
Comment (pbartholomew08) [inline src/linear_solvers/custom_matrix.f90]: intent in, out, inout?

File: src/linear_solvers/custom_matrix.f90 [hunk 3]
```diff
@@ -10,39 +12,100 @@ module custom_matrix
...
+    allocate(matrix%values(num_rows * values_per_row))
     matrix%values(:) = 0.0
 
-    allocate(matrix%columns(rows * values_per_row))
+    allocate(matrix%columns(num_rows * values_per_row))
     matrix%columns(:) = 0
 
   end subroutine
 
+  subroutine get_offsets(row, start, end, matrix)
```
Comment (pbartholomew08) [inline src/linear_solvers/custom_matrix.f90]: argument order is reversed

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]

File: src/linear_solvers/custom_matrix.f90 (hunk 3 above)
Comment (justsz) [reply src/linear_solvers/custom_matrix.f90]: Will fix.

File: src/linear_solvers/custom_matrix.f90 (hunk 2 above)
Comment (justsz) [reply src/linear_solvers/custom_matrix.f90]: Will fix.

File: src/linear_solvers/custom_matrix.f90 (hunk 1 above)
Comment (justsz) [reply src/linear_solvers/custom_matrix.f90]: Will fix.

**Review by pbartholomew08 — APPROVED** [review]

**Pull request merged**